### ISSUE TITLE
sample is confusing

### ISSUE THREAD

**Issue by usman-ghauri** [timehop/sticky-headers-recyclerview#118]
i've implemented adapters as suggested in sample and its still not showing anything on screen

**Comment by Nizomjon1994** [comment]
Did u create view for header ??

**Comment by usman-ghauri** [comment]
yeah i did. 

`/**
 * Created by ghauri on 4/1/2016.
 */
public class DriveHeaderAdapter extends DriveItemInterfaceAdapter<RecyclerView.ViewHolder>
        implements StickyRecyclerHeadersAdapter<RecyclerView.ViewHolder> {

    public DriveHeaderAdapter(Context context, String accessToken) {
        this.mContext = context;
        this.mAccessToken = accessToken;
    }

    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.photos_item_thumb_view, parent, false);
        return new RecyclerView.ViewHolder(view) {
        };
    }

    @Override
    public void onBindViewHolder(RecyclerView.ViewHolder holder, int position) {
        DriveItem driveItem = getItem(position);
        TextView ivItemTitle = (TextView) holder.itemView.findViewById(R.id.tv_item_title);
        ImageView ivItemIcon = (ImageView) holder.itemView.findViewById(R.id.iv_item_icon);
        ImageView ivItemThumb = (ImageView) holder.itemView.findViewById(R.id.iv_item_thumb);
        if (driveItem != null) {
            //Setting text view title
            ivItemTitle.setText(driveItem.getTitle());
            if (driveItem.getMimeType().equalsIgnoreCase("image/jpeg") || driveItem.getMimeType().equalsIgnoreCase("video/mp4"))
                PicassoUtils.picassoLoadImageFromUrl(mContext, ivItemThumb,
                        driveItem.getThumbnailLink(), R.drawable.file_type_small_default, true, true, 0);
            else
                PicassoUtils.picassoLoadImageFromUrl(mContext, photoItemViewHolder.ivItemThumb,
                        driveItem.getIconLink(), R.drawable.file_type_small_default, true, true, 0);

            if (driveItem.getMimeType().equalsIgnoreCase("image/jpeg"))
                photoItemViewHolder.ivItemIcon.setImageResource(R.drawable.file_type_small_image);
            else
                PicassoUtils.picassoLoadImageFromUrl(mContext, photoItemViewHolder.ivItemIcon,
                        driveItem.getIconLink(), R.drawable.file_type_small_default, true, true, 0);

            //Handle click event on both title and image click
            photoItemViewHolder.tvItemTitle.setOnClickListener(clickListener);
            photoItemViewHolder.tvItemTitle.setTag(photoItemViewHolder);
            photoItemViewHolder.ivItemThumb.setOnClickListener(clickListener);
            photoItemViewHolder.ivItemThumb.setTag(photoItemViewHolder);
            //adjust imageview for different screen size
            android.view.ViewGroup.LayoutParams layoutParams = photoItemViewHolder.ivItemThumb.getLayoutParams();
            layoutParams.height = (int) (MainActivity.SCREEN_HEIGHT * 0.25);
            photoItemViewHolder.ivItemThumb.setLayoutParams(layoutParams);
        }     
        //TextView textView = (TextView) holder.itemView;
        //textView.setText(getItem(position));
    }

    /**
     * Get the ID of the header associated with this item.  For example, if your headers group
     * items by their first letter, you could return the character representation of the first letter.
     * Return a value < 0 if the view should not have a header (like, a header view or footer view)
     *
     * @param position
     * @return
     */
    @Override
    public long getHeaderId(int position) {
        if (position == 0) {
            return -1;
        } else {
            return getItem(position).getMimeType().charAt(0);
        }
    }

    @Override
    public RecyclerView.ViewHolder onCreateHeaderViewHolder(ViewGroup parent) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.header_view_recycler, parent, false);
        return new RecyclerView.ViewHolder(view) {
        };
    }

    @Override
    public void onBindHeaderViewHolder(RecyclerView.ViewHolder holder, int position) {
        DriveItem driveItem = getItem(position);

        TextView textView = (TextView) holder.itemView;
        textView.setText(String.valueOf(driveItem.getTitle()));
        //holder.itemView.setBackgroundColor(getRandomColor());
    }

    @Override
    public int getItemCount() {
        return 0;
    }

    private int getRandomColor() {
        SecureRandom secureRandom = new SecureRandom();
        return Color.HSVToColor(150, new float[]{
                secureRandom.nextInt(359), 1, 1
        });
    }

}`

**Comment by usman-ghauri** [comment]
Yes i did

'
public class DriveHeaderAdapter extends DriveItemInterfaceAdapter<RecyclerView.ViewHolder>
        implements StickyRecyclerHeadersAdapter<RecyclerView.ViewHolder> {

    public DriveHeaderAdapter(Context context, String accessToken) {
        this.mContext = context;
        this.mAccessToken = accessToken;
    }

    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.photos_item_thumb_view, parent, false);
        return new RecyclerView.ViewHolder(view) {
        };
    }

    @Override
    public void onBindViewHolder(RecyclerView.ViewHolder holder, int position) {
        DriveItem driveItem = getItem(position);
        TextView ivItemTitle = (TextView) holder.itemView.findViewById(R.id.tv_item_title);
        ImageView ivItemIcon = (ImageView) holder.itemView.findViewById(R.id.iv_item_icon);
        ImageView ivItemThumb = (ImageView) holder.itemView.findViewById(R.id.iv_item_thumb);
        if (driveItem != null) {
            //Setting text view title
            ivItemTitle.setText(driveItem.getTitle());
            if (driveItem.getMimeType().equalsIgnoreCase("image/jpeg") || driveItem.getMimeType().equalsIgnoreCase("video/mp4"))
                PicassoUtils.picassoLoadImageFromUrl(mContext, ivItemThumb,
                        driveItem.getThumbnailLink(), R.drawable.file_type_small_default, true, true, 0);
            else
                PicassoUtils.picassoLoadImageFromUrl(mContext, photoItemViewHolder.ivItemThumb,
                        driveItem.getIconLink(), R.drawable.file_type_small_default, true, true, 0);

            if (driveItem.getMimeType().equalsIgnoreCase("image/jpeg"))
                photoItemViewHolder.ivItemIcon.setImageResource(R.drawable.file_type_small_image);
            else
                PicassoUtils.picassoLoadImageFromUrl(mContext, photoItemViewHolder.ivItemIcon,
                        driveItem.getIconLink(), R.drawable.file_type_small_default, true, true, 0);

            //Handle click event on both title and image click
            photoItemViewHolder.tvItemTitle.setOnClickListener(clickListener);
            photoItemViewHolder.tvItemTitle.setTag(photoItemViewHolder);
            photoItemViewHolder.ivItemThumb.setOnClickListener(clickListener);
            photoItemViewHolder.ivItemThumb.setTag(photoItemViewHolder);
            //adjust imageview for different screen size
            android.view.ViewGroup.LayoutParams layoutParams = photoItemViewHolder.ivItemThumb.getLayoutParams();
            layoutParams.height = (int) (MainActivity.SCREEN_HEIGHT * 0.25);
            photoItemViewHolder.ivItemThumb.setLayoutParams(layoutParams);
        }     
        //TextView textView = (TextView) holder.itemView;
        //textView.setText(getItem(position));
    }

    /**
     * Get the ID of the header associated with this item.  For example, if your headers group
     * items by their first letter, you could return the character representation of the first letter.
     * Return a value < 0 if the view should not have a header (like, a header view or footer view)
     *
     * @param position
     * @return
     */
    @Override
    public long getHeaderId(int position) {
        if (position == 0) {
            return -1;
        } else {
            return getItem(position).getMimeType().charAt(0);
        }
    }

    @Override
    public RecyclerView.ViewHolder onCreateHeaderViewHolder(ViewGroup parent) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.header_view_recycler, parent, false);
        return new RecyclerView.ViewHolder(view) {
        };
    }

    @Override
    public void onBindHeaderViewHolder(RecyclerView.ViewHolder holder, int position) {
        DriveItem driveItem = getItem(position);

        TextView textView = (TextView) holder.itemView;
        textView.setText(String.valueOf(driveItem.getTitle()));
        //holder.itemView.setBackgroundColor(getRandomColor());
    }

    @Override
    public int getItemCount() {
        return 0;
    }

    private int getRandomColor() {
        SecureRandom secureRandom = new SecureRandom();
        return Color.HSVToColor(150, new float[]{
                secureRandom.nextInt(359), 1, 1
        });
    }

}`

**Comment by miguelespinoza** [comment]
Also was experiencing this issue, have the similar setup as @usman-ghauri. Turns out that for me. I was adding my values to the ArrayList that was associated with my adapter. 

example: 
`list.addAll(items)`

changed to:
`adapter.addAll(items)`

then:
`adapter.notifyDataSetChanged()`

Voila! You can see your UI now. 

Don't know if this is a bug. b/c you should be able to `list.addAll(items)` and then notify your adapter

**Comment by usman-ghauri** [comment]
thanks @miguelespinoza im going to test that and confirm if it solves this issue.

**Comment by jeanvallon** [comment]
Hello, I have the same issue as @usman-ghauri 
The solution of @miguelespinoza didn't work for me. I am already adding items with a method "addData()" I created in my custom adapter.

**Comment by usman-ghauri** [comment]
Ok what i did is in header adapter where i bind view of header and items , i added called super addall function from the abstract class they defined 
        super.addAll(driveItemList);
and then it started displaying items but there is no header still.

now the question is, on what basis is it attaching header? 

is it this function 

    @Override
    public long getHeaderId(int position) {
        if (position == 0) {
            return -1;
        } else {
            return getItem(position).getMimeType().charAt(0);
        }
    }

because in this function im trying to show header of file type in header. 

also the view is in accurate too im using same layout file for both adapters 
This screenshot is with this adapter and imports.
![screenshot_2016-04-06-17-21-52](https://cloud.githubusercontent.com/assets/12395698/14316319/d322c900-fc1b-11e5-8523-a3668a9517af.png)

and this screenshot is from my adapter 
![screenshot_2016-04-06-17-22-19](https://cloud.githubusercontent.com/assets/12395698/14316360/032d9878-fc1c-11e5-82b3-30db2fb60655.png)


Both these adapter uses same layout file but have different views. probably because of that header view?

**Comment by jeanvallon** [comment]
I don't think that the the header is only attached thanks to the function getHeaderId because what I did was :
```
@Override
    public long getHeaderId(int position) {
        return 1;
    }
@Override
    public void onBindHeaderViewHolder(RecyclerView.ViewHolder holder, int position) {
        TextView textView = (TextView) holder.itemView;
        textView.setText("Posts");
    }
```
And I don't have the header with the text "Posts" displayed on top.

**Comment by jeanvallon** [comment]
Actually the function getHeaderId() isn't event called. And onCreateViewHolder() neither.

**Comment by jeanvallon** [comment]
Okay I found the reason why I couldn't see my header !
First of all, it is compulsory to add the following decorations :
```
final StickyRecyclerHeadersDecoration headersDecor = new StickyRecyclerHeadersDecoration(placeUsersAdapter);
        peopleRv.addItemDecoration(headersDecor);
```

But even with that I couldn't see the text of the header (I could see its background though).
So I removed the line "singleLine=true" I had in the textview of the header, and then the text was finally visible!
I am so happy... This is the best day of my life.

**Comment by jeanvallon** [comment]
@usman-ghauri you should complete the title of the issue with "can't see header" or something like that

**Comment by usman-ghauri** [comment]
@jeanvallon yeah i didn't decorated header either. i have to check that out and test it on my project. thanks for follow up
